catalog/dags/data_refresh/reporting.py

```diff
@@ -17,21 +17,23 @@ def report_status(media_type: str, message: str, dag_id: str):
     return message
 
 
-def report_record_difference(before: str, after: str, media_type: str, dag_id: str):
-    before = int(before)
-    after = int(after)
-    count_diff = after - before
-    percent_diff = (count_diff / before) * 100
-    # Note for formatting:
-    # '+' - number will always have a sign in front of it
-    # ',' - number is comma separated
-    # '.' - number is a float
+def report_record_difference(before: dict, after: dict, media_type: str, dag_id: str):
+    all_keys = before.keys() | after.keys()
+    total_before = sum(before.values())
+    total_after = sum(after.values())
+    count_diff = total_after - total_before
+    percent_diff = (count_diff / total_before) * 100
```

Comment:
Note that when we don't have an index, this fails:
```
[2024-04-13, 01:00:27 UTC] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 439, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/catalog/dags/data_refresh/reporting.py", line 25, in report_record_difference
    percent_diff = (count_diff / total_before) * 100
ZeroDivisionError: division by zero
```

I was able to use the following locally with success, but feel free to take a different approach:
```suggestion
    if total_before == 0:
        percent_diff = float("inf")
    else:
        percent_diff = (count_diff / total_before) * 100
```